src/NUnitFramework/framework/Attributes/RandomAttribute.cs

```diff
@@ -221,6 +243,7 @@ private bool WeConvert(Type sourceType, Type targetType)
         abstract class RandomDataSource : IParameterDataSource
         {
             public Type DataType { get; protected set; }
+            public bool Distinct { get; set; }
```

Comment:
I would have preferred making this `public bool Distinct { get; }` and only settable through the constructor, so that the same instance can't be used for both distinct and non-distinct generation. It's not entirely clear what the behavior should be if you turn this off, generate some numbers, and turn it back on again and generate more. I don't know how big of an issue this is though.